examples/nrf/src/bin/pwm.rs

```diff
@@ -7,95 +7,42 @@ mod example_common;
-    354, 356, 359, 361, 364, 367, 370, 373, 376, 379, 383, 386, 390, 394, 398, 402, 406, 410, 415,
-    419, 424, 429, 434, 439, 444, 450, 455, 461, 467, 473, 479, 485, 492, 498, 505, 512, 519, 526,
-    534, 541, 549, 557, 565, 574, 582, 591, 599, 609, 618, 627, 637, 647, 657, 667, 677, 688, 699,
-    710, 721, 733, 744, 756, 768, 781, 793, 806, 819, 833, 846, 860, 874, 888, 903, 918, 933, 948,
-    964, 980, 996, 1012, 1029, 1046, 1063, 1081, 1099, 1117, 1135, 1154, 1173, 1192, 1212, 1232,
-    1252, 1273, 1294, 1315, 1336, 1358, 1380, 1403, 1426, 1449, 1472, 1496, 1520, 1545, 1570, 1595,
-    1620, 1646, 1673, 1699, 1726, 1754, 1781, 1810, 1838, 1867, 1896, 1926, 1956, 1986, 2017, 2048,
-    2080, 2112, 2144, 2177, 2210, 2244, 2278, 2312, 2347, 2382, 2418, 2454, 2491, 2528, 2565, 2603,
-    2641, 2680, 2719, 2758, 2798, 2839, 2879, 2921, 2962, 3005, 3047, 3090, 3134, 3178, 3222, 3267,
-    3312, 3358, 3404, 3451, 3498, 3546, 3594, 3642, 3691, 3741, 3791, 3841, 3892, 3943, 3995, 4047,
-    4100, 4153, 4207, 4261, 4316, 4371, 4426, 4482, 4539, 4596, 4653, 4711, 4769, 4828, 4888, 4947,
-    5008, 5068, 5130, 5191, 5253, 5316, 5379, 5443, 5507, 5571, 5636, 5702, 5767, 5834, 5901, 5968,
-    6036, 6104, 6172, 6241, 6311, 6381, 6451, 6522, 6594, 6665, 6738, 6810, 6883, 6957, 7031, 7105,
-    7180, 7255, 7331, 7407, 7484, 7561, 7638, 7716, 7794, 7873, 7952, 8031, 8111,
-];
-
 #[embassy::main]
 async fn main(_spawner: Spawner, p: Peripherals) {
-    let pwm = Pwm::new(p.PWM0, p.P0_13, p.P0_14, p.P0_16, p.P0_15);
-    pwm.set_prescaler(Prescaler::Div1);
+    let pwm = Pwm::new(p.PWM0, p.P0_05, NoPin, NoPin, NoPin);
+    // sg90 microervo requires 50hz or 20ms period
+    // set_period can only set down to 125khz so we cant use it directly
+    // Div128 is 125khz or 0.000008s or 0.008ms, 20/0.008 is 2500 is top
+    pwm.set_prescaler(Prescaler::Div128);
+    pwm.set_max_duty(2500);
     info!("pwm initialized!");
 
-    let mut i = 0;
+    Timer::after(Duration::from_millis(5000)).await;
+
+    // 1ms 0deg (1/.008=125), 1.5ms 90deg (1.5/.008=187.5), 2ms 180deg (2/.008=250),
     loop {
-        i += 1;
-        pwm.set_duty(0, DUTY[i % 1024]);
-        pwm.set_duty(1, DUTY[(i + 256) % 1024]);
-        pwm.set_duty(2, DUTY[(i + 512) % 1024]);
-        pwm.set_duty(3, DUTY[(i + 768) % 1024]);
-        Timer::after(Duration::from_millis(3)).await;
+        info!("45 deg");
+        pwm.set_duty(0, 2500 - 156);
+        Timer::after(Duration::from_millis(5000)).await;
+
+        info!("90 deg");
+        pwm.set_duty(0, 2500 - 187);
+        Timer::after(Duration::from_millis(5000)).await;
+
+        info!("135 deg");
+        pwm.set_duty(0, 2500 - 218);
+        Timer::after(Duration::from_millis(5000)).await;
+
+        info!("180 deg");
+        pwm.set_duty(0, 2500 - 250);
+        Timer::after(Duration::from_millis(5000)).await;
+
+        info!("0 deg");
+        pwm.set_duty(0, 2500 - 125);
+        Timer::after(Duration::from_millis(5000)).await;
```

Comment:
Wdyt about minimising the changes to this example? I thought what we had was quite straightforward, and it’d be interesting to make a direct comparison.

Reply:
Yeah I dont much care what the examples are, this isnt final. However I disagree thats a good example. Its attempting to create a tight sequence which is literally what the new api would be for, with a loose timing scheduler which is a bad idea.

The old example also cant work anymore because it was relying on a the .refresh field to be stretching out the sequence buffer which is a very advanced api. if that sin table is recreated the old example could be returned.

Reply:
https://github.com/embassy-rs/embassy/pull/455/files#diff-c1b3acd7a9372df9109e152f99037a9d9f1c6db11ec955ca1792830adb434de3L84

Reply:
I threw a commit on which restores the example by adding a new api set_time_stretch. Not sure what other drives elsewhere call it. Its the 'refresh' register on nrf. Now other commenters have said they dont want more api surface necessarily, so Ill leave that up to someone else. But having it set in the constructor with no way to change as it is in the upstream wasnt great

Reply:
The old PWM api and example didn't need `refresh`.  It just plays a 1-sample sequence. When the sequence is done playing (which is immediately), the last duty keeps playing "forever". This is how you "set duty".

> After the last value in the sequence has been loaded and started executing, a SEQEND[n] event is generated. The PWM generation will then continue with the last loaded value.

Reply:
Ok maybe this is the last clarification I needed. If the sequence is 1 long, it plays forever, but if the sequence is larger than 1, it stops?

Reply:
no rather. 

it always the last value in the register.. so even on a 5 times sequence it only plays the sequence five times and also leaves the last value in the register in that case